tests_and_analysis/test/euphonic_test/test_plot.py

```diff
@@ -0,0 +1,332 @@
+
+    @pytest.mark.parametrize('spec1_units', [('angstrom^-1', 'hartree'),
+                                             ('bohr^-1', 'meV')])
+    def test_plot_badunits(self, spec1_units, band_segments):
+        spec1, spec2 = band_segments
+        spec1.x_data_unit, spec1.y_data_unit = spec1_units
+        spec2.x_data_unit, spec2.y_data_unit = ('angstrom^-1', 'meV')
+        with pytest.raises(ValueError):
+            plot_1d([spec1, spec2])
+
+    @pytest.mark.parametrize('kwargs', [{'labels': ['band A', 'band B']},
+                                        {'x_label': 'X_LABEL',
+                                         'y_label': 'Y_LABEL',
+                                         'title': 'TITLE'}])
+    def test_plot_multi(self, mocker, band_segments, kwargs):
+
+        suptitle = mocker.patch.object(matplotlib.figure.Figure, 'suptitle')
+
+        fig = plot_1d(band_segments, **kwargs)
+
+        if 'labels' in kwargs:
+            legend = fig.axes[0].get_legend()
+            assert legend is not None
+
+            for text, label in zip(legend.get_texts(), kwargs['labels']):
+                assert text.get_text() == label
+
+        if 'x_label' in kwargs:
+            assert fig.axes[-1].get_xlabel() == kwargs['x_label']
+        if 'y_label' in kwargs:
+            assert fig.axes[-1].get_ylabel() == kwargs['y_label']
+        if 'title' in kwargs:
+            suptitle.assert_called_once_with(kwargs['title'])
+
+        for ax in fig.axes[:-1]:
+            if 'y_min' in kwargs:
+                assert ax.get_ylim()[0] == pytest.approx(kwargs.get('y_min'))
+            if 'y_max' in kwargs:
+                assert ax.get_ylim()[0] == pytest.approx(kwargs.get('y_max'))
+
+
+@pytest.mark.unit
+class TestPlotDispersion:
+    @pytest.fixture
+    def bands(self, mocker):
+        spectrum = mocker.MagicMock(spec_set=Spectrum1DCollection)
+        spectrum.split.return_value = [1, 2]
+
+        return spectrum
+
+    @pytest.fixture
+    def phonons(self, mocker, bands):
+        from euphonic import QpointPhononModes
+        modes = mocker.MagicMock(spec_set=QpointPhononModes)
+        modes.get_dispersion.return_value = bands
+
+        return modes
+
+    @pytest.mark.parametrize('kwargs',
+                             [{'btol': 5., 'kwarg1': 1, 'kwarg2': 2},
+                              {'kwarg1': 2},
+                              {}])
+    def test_plot_dispersion(self, mocker, phonons, bands, kwargs):
+        import euphonic.plot
+        mocker.patch('euphonic.plot.plot_1d', return_value=None)
+
+        plot_dispersion(phonons, **kwargs)
+
+        assert phonons.get_dispersion.called
+        if 'btol' in kwargs:
+            bands.split.assert_called_once_with(btol=kwargs['btol'])
+        else:
+            bands.split.assert_called_once_with(btol=10.)
+
+        euphonic.plot.plot_1d.assert_called_once_with(
+            [1, 2],
+            **{k: v for k, v in kwargs.items() if k != 'btol'})
```

Comment:
This has also made me realise that `get_dispersion` is actually not tested in `test_qpoint_phonon_modes`, but it really should be. And should `StructureFactor` also have a `get_dispersion` method? It contains all the information needed for it and might be something people would want to do

Reply:
Yes to both! But preferably not in this PR...

Reply:
Are we going to require `StructureFactor.get_dispersion` for this feature branch? Or open a separate Issue?

Reply:
Perhaps a separate issue? There might need to be some thought over whether we should just duplicate the `get_dispersion` code or have some kind of `QpointPhononFrequencies` superclass or something else entirely

Reply:
Done! #117